src/isomorphic/classic/element/ReactCurrentOwner.js

```diff
@@ -18,13 +18,11 @@
  * currently being constructed.
  */
 var ReactCurrentOwner = {
-
-  /**
-   * @internal
-   * @type {ReactComponent}
-   */
   current: null,
 
+  inject: function(InjectedCurrentOwner) {
+    ReactCurrentOwner = InjectedCurrentOwner;
```

Comment:
The `InjectedCurrentOwner` is just `{current: null}` which works by reference, but does feel slightly hackish.